samples: bluetooth: fast_pair: clean up leftover nRF54H20 DK configuration (#28359)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

NCSDK-34821: Fast Pair samples do not build or run due to the IronSide SE migration
  The :ref:`fast_pair_input_device` and :ref:`fast_pair_locator_tag` samples are currently incompatible with the latest |NCS| transition to the IronSide SE solution - bundle 22.1.0 and higher for the ``nrf54h20dk/nrf54h20/cpuapp`` board target.
  As a result, you cannot use these samples with that board target in any of the affected releases.

>   The support for the ``nrf54h20dk/nrf54h20/cpuapp`` board target has been removed after the v3.3.0 release.

Reviewer: ```suggestion
  The support for the ``nrf54h20dk/nrf54h20/cpuapp`` board target has been removed after the |NCS| v3.3.0 release.
```

>   The support for the ``nrf54h20dk/nrf54h20/cpuapp`` board target has been removed after the |NCS| v3.3.0 release.

Reviewer 2: Support has been removed in the NCS v3.1.0

This statement appears to contradict what the changelog states.

```suggestion

  Although the ``nrf54h20dk/nrf54h20/cpuapp`` board target has not been functional with the affected samples since the IronSide SE transition in the |NCS| v3.1.0 release, the samples continued to ship with leftover configuration files and documentation that described the target as supported.
  The |NCS| v3.4.0 release removes those leftovers, aligns the documentation with the actual state, officially drops support for this board target, and closes this known issue for subsequent releases.
```

NCSDK-34821: Fast Pair samples do not build or run due to the IronSide SE migration
  The :ref:`fast_pair_input_device` and :ref:`fast_pair_locator_tag` samples are currently incompatible with the latest |NCS| transition to the IronSide SE solution - bundle 22.1.0 and higher for the ``nrf54h20dk/nrf54h20/cpuapp`` board target.
  As a result, you cannot use these samples with that board target in any of the affected releases.

>   Although the ``nrf54h20dk/nrf54h20/cpuapp`` board target has not been functional with the affected samples since the IronSide SE transition in the |NCS| v3.1.0 release, the samples continued to ship with leftover configuration files and documentation that described the target as supported.

Reviewer 2: ```suggestion

  Although the ``nrf54h20dk/nrf54h20/cpuapp`` board target has not been functional with the affected samples since the IronSide SE transition in the |NCS| v3.1.0 release, the samples continued to ship with leftover configuration files and documentation that described the target as supported.
```

NCSDK-34821: Fast Pair samples do not build or run due to the IronSide SE migration
  The :ref:`fast_pair_input_device` and :ref:`fast_pair_locator_tag` samples are currently incompatible with the latest |NCS| transition to the IronSide SE solution - bundle 22.1.0 and higher for the ``nrf54h20dk/nrf54h20/cpuapp`` board target.
  As a result, you cannot use these samples with that board target in any of the affected releases.
  Although the ``nrf54h20dk/nrf54h20/cpuapp`` board target has not been functional with the affected samples since the IronSide SE transition in the |NCS| v3.1.0 release, the samples continued to ship with leftover configuration files and documentation that described the target as supported.

>   The |NCS| v3.4.0 release removes those leftovers, aligns the documentation with the actual state, officially drops support for this board target, and closes this known issue for subsequent releases.

Reviewer 2: ```suggestion

  Although the ``nrf54h20dk/nrf54h20/cpuapp`` board target has not been functional with the affected samples since the IronSide SE transition in the |NCS| v3.1.0 release, the samples continued to ship with leftover configuration files and documentation that described the target as supported.
  The |NCS| v3.4.0 release removes those leftovers, aligns the documentation with the actual state, officially drops support for this board target, and closes this known issue for subsequent releases.
```
